src/cclib/parser/molproparser.py

```diff
@@ -28,20 +29,25 @@ def create_atomic_orbital_names(orbitals):
         'P': ['x', 'y', 'z', '2px', '2py', '2pz'],
     }
 
-    # Although we could write out all names for the other subshells, it is better
-    # to generate them if we need to expand further, since the number of functions quickly
-    # grows and there are both Cartesian and spherical variants to consider.
-    # For D orbitals, the Cartesian functions are xx, yy, zz, xy, xz and yz, and the
-    # spherical ones are called 3d0, 3d1-, 3d1+, 3d2- and 3d2+. For F orbitals, the Cartesians
-    # are xxx, xxy, xxz, xyy, ... and the sphericals are 4f0, 4f1-, 4f+ and so on.
+    # Although we could write out all names for the other subshells, it is
+    # better # to generate them if we need to expand further,
+    # since the number of functions quickly grows and there are both
+    # Cartesian and spherical variants to consider.
+    # For D orbitals, the Cartesian functions are xx, yy, zz, xy, xz and yz,
+    # and the # spherical ones are called 3d0, 3d1-, 3d1+, 3d2- and 3d2+.
+    # For F orbitals, the Cartesians are xxx, xxy, xxz, xyy, ...
+    # and the sphericals are 4f0, 4f1-, 4f+ and so on.
     for i, orb in enumerate(orbitals):
 
         # Cartesian can be generated directly by combinations.
-        cartesian = list(map(''.join, list(itertools.combinations_with_replacement(['x', 'y', 'z'], i+2))))
+        cartesian = [
+            ''.join(x) for x in
+            itertools.combinations_with_replacement(['x', 'y', 'z'], i + 2)]
```

Comment:
I'm ok with removing `map`, but see the later dangling delimiter comment.

One additional way it could be shortened is to stick a `from itertools import combinations_with_replacement` at the top of the file.